### PR TITLE
[WIP] Add thanos v0.5.0 support for release-0.1 branch

### PR DESCRIPTION
**Pull request by curantes**
This PR is similar to #140 but is targeted release-0.1 branch to support
Kubernetes version < v1.14.0

### CODE REVIEW

File: jsonnet/kube-prometheus/kube-prometheus-thanos.libsonnet [hunk 1]
```diff
@@ -5,7 +5,7 @@ local servicePort = k.core.v1.service.mixin.spec.portsType;
 {
   _config+:: {
     versions+:: {
-      thanos: 'v0.3.2',
+      thanos: 'v0.5.0',
```
Comment (brancz) [inline jsonnet/kube-prometheus/kube-prometheus-thanos.libsonnet]: have you tried this? I think this needs an update of the prometheus-operator as well to work

Comment (curantes) [reply jsonnet/kube-prometheus/kube-prometheus-thanos.libsonnet]: No its not fully tested yet. I am in the process of doing this now. Maybe I should add "WIP" into headline or close this until I have fully verified this. Sorry this is my first PR here. What do do you prefer?

Comment (brancz) [reply jsonnet/kube-prometheus/kube-prometheus-thanos.libsonnet]: Feel free to leave open with WIP just let us know when you've verified it and want us to review :)

Comment (curantes) [reply jsonnet/kube-prometheus/kube-prometheus-thanos.libsonnet]: yes will do, thanks!

**Comment by paulfantom** [comment]
I am closing this due to staleness. Please reopen if you still want to work on this.

**Pull request closed without merging**